ar_track_alvar/CMakeLists.txt

```diff
@@ -1,183 +1,187 @@
-cmake_minimum_required(VERSION 2.8.3)
+cmake_minimum_required(VERSION 3.5)
 project(ar_track_alvar)
 
-set(MSG_DEPS
-    ar_track_alvar_msgs
-    std_msgs
-    sensor_msgs
-    geometry_msgs
-    visualization_msgs)
-
-find_package(catkin COMPONENTS
-        genmsg
-        roscpp
-        tf
-        tf2
-        image_transport
-        resource_retriever
-        cv_bridge
-        pcl_ros
-        pcl_conversions
-        message_generation
-        ${MSG_DEPS}
-        dynamic_reconfigure
-        cmake_modules
-        REQUIRED)
+find_package(ament_cmake REQUIRED)
+find_package(ament_cmake_auto REQUIRED)
+ament_auto_find_build_dependencies()
 
-find_package(Eigen3 REQUIRED)
 find_package(OpenCV REQUIRED)
-find_package(TinyXML REQUIRED)
-
-include(CheckCXXCompilerFlag)
-CHECK_CXX_COMPILER_FLAG("-std=c++11" COMPILER_SUPPORTS_CXX11)
-CHECK_CXX_COMPILER_FLAG("-std=c++0x" COMPILER_SUPPORTS_CXX0X)
-if(COMPILER_SUPPORTS_CXX11)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
-elseif(COMPILER_SUPPORTS_CXX0X)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++0x")
-else()
-        message(STATUS "The compiler ${CMAKE_CXX_COMPILER} has no C++11 support. Please use a different C++ compiler.")
+if(NOT DEFINED CMAKE_SUPPRESS_DEVELOPER_WARNINGS)
+     set(CMAKE_SUPPRESS_DEVELOPER_WARNINGS 1 CACHE INTERNAL "No dev warnings")
 endif()
 
-cmake_policy(SET CMP0046 OLD)
-
-# dynamic reconfigure support
-generate_dynamic_reconfigure_options(cfg/Params.cfg)
-
-catkin_package(
-  INCLUDE_DIRS include
-  LIBRARIES ar_track_alvar
-  CATKIN_DEPENDS
-        ar_track_alvar_msgs
-        std_msgs
-        roscpp
-        tf
-        tf2
-        message_runtime
-        image_transport
-        sensor_msgs
-        geometry_msgs
-        visualization_msgs
-        resource_retriever
-        cv_bridge
-        pcl_ros
-        pcl_conversions
-        dynamic_reconfigure
+find_package(rclcpp REQUIRED)
+find_package(tf2_ros REQUIRED)
+find_package(tf2 REQUIRED)
+find_package(image_transport REQUIRED)
+find_package(resource_retriever REQUIRED)
+find_package(cv_bridge REQUIRED)
+find_package(perception_pcl  REQUIRED)
+find_package(pcl_ros)
+find_package(pcl_conversions REQUIRED)
+find_package(ar_track_alvar_msgs REQUIRED)
+find_package(std_msgs REQUIRED)
+find_package(sensor_msgs REQUIRED)
+find_package(tf2_geometry_msgs REQUIRED)
+find_package(geometry_msgs REQUIRED)
+find_package(rosbag2_bag_v2_plugins REQUIRED)
+find_package(visualization_msgs REQUIRED)
+find_package(PCL REQUIRED QUIET COMPONENTS common io)
+find_package(Eigen3 REQUIRED)
+find_package(tinyxml_vendor REQUIRED)
+
+include_directories(include)
+
+# Kinect filtering code
+set(KINECT_FILTERING_TARGETS kinect_filtering medianFilter)
+set(ALVAR_TARGETS ar_track_alvar individualMarkers individualMarkersNoKinect trainMarkerBundle findMarkerBundles findMarkerBundlesNoKinect createMarker)
+
+set(dependencies
+  OpenCV
+  tf2_ros 
+  tf2
+  pcl_ros
```

Comment:
I tried run `rosdep install --from-paths src --ignore-src --rosdistro foxy -yr` during installation, and this is not specified in `package.xml`. Also realized that `pcl_ros` is not available as debian in foxy. 

Anyway, we can also remove this since that `pcl_ros` is not used anyway in the code.

Reply:
Thanks @youliangtan I made the changes you suggested